daemon: Fail init if requirements for BPF masquerade are not met

pippolo84 opened the pull request:
In case the requirements for BPF masquerade are not met, fail the daemon initialization instead of silently fall back to iptables based masquerading.

This is needed because the iptables cell does not support runtime configuration changes.

Fixes: https://github.com/cilium/cilium/issues/29663

pippolo84 committed:
daemon: Fail init if requirements for BPF masquerade are not met

In case the requirements for BPF masquerade are not met, fail the daemon
initialization instead of silently fall back to iptables based
masquerading.

This is needed because the iptables cell does not support runtime
configuration changes.

Fixes: a6a2b73f4f ("iptables: Add a cell for iptables config manager")

Signed-off-by: Fabio Falzoi <fabio.falzoi@isovalent.com>

giorio94 reviewed (changes requested):
Thanks! 

There's another occurrence which also sets `EnableBPFMasquerade` to false a few lines below. It may not be affected (as in that case `MasqueradingEnabled()` is false), but it is likely worth to also address that one, so that the value is effectively never mutated.

learnitall reviewed (approved):
Upgrade notes look good!

giorio94 reviewed (approved):
Thanks!

pippolo84 committed:
test: Disable BPF masquerade in fromEntities policies tests

The previous commit removed the silent fallback to iptables based
masquerading if the requirements for BPF masquerading are not met.

Since having both enable-ipv4-masquerade and enable-ipv6-masquerade set
to false is incompatible with BPF masquerading, the ginkgo tests that
need this configuration now have to explicitly set iptables based
masquerading.

Signed-off-by: Fabio Falzoi <fabio.falzoi@isovalent.com>
